libnmstate/nm/connection.py

```diff
@@ -249,34 +258,77 @@ def _get_activation_metadata(self):
 
         return activation_type, activation_object
 
-    def waitfor_active_connection_async(self, ac):
-        ac.handlers.add(
```

Comment:
If I understood correctly, you have changed the event driven handling to a poll mode. Sounds very costly.
Can't we just register a timeout for the handler of "state-changed"? I just find it strange that we need to use poll mode here in order to handle the collision.

Reply:
As I said in https://github.com/nmstate/nmstate/pull/478#discussion_r315980457 the active connection might be became None hence its signal/handler cannot be trusted.